features/data_converter/empty/feature.ts

```diff
@@ -0,0 +1,44 @@
+import { JSONToPayload } from '@temporalio/common/lib/proto-utils';
+import { Feature } from '@temporalio/harness';
+import * as assert from 'assert';
+import { ApplicationFailure } from '@temporalio/common';
+import expectedPayload from './payload.json';
+import * as wf from '@temporalio/workflow';
+
+const activitiesImpl = {
+  async activity(input: any): Promise<void> {
+    if (input != undefined) {
+      throw new ApplicationFailure('Activity input should be undefined', null, true, undefined, undefined);
```

Comment:
You can just throw an `Error` for retryable or use the `ApplicationFailure.nonRetryable` factory method instead of the constructor.